src/DateTimeFormatter.php

```diff
@@ -78,9 +57,9 @@ public function filter(mixed $value): mixed
     /**
      * Normalize the provided value to a formatted string
      *
-     * @return string|mixed
+     * @throws Exception
      */
-    protected function normalizeDateTime(mixed $value)
+    protected function normalizeDateTime(mixed $value): mixed
```

Comment:
@gsteel  Here i tried another aproach, instead of checking for different types (string, int..) and returning the values that cannot be converted to a DateTimeImmutable, i let the DateTimeImmutable try to create the date object and to catch the exception and throw an InvalidArgumentException
Also, i added the timezone support here

Reply:
I've added a review comment about this - filters are supposed to attempt to filter only arguments that it understands, so types like `null`, `array`, `float`, `stdClass` etc should be returned un-filtered rather than causing an exception.